aten/src/ATen/native/cuda/SoftMax.cu

```diff
@@ -899,8 +899,13 @@ Tensor host_softmax_backward(const Tensor &grad_, const Tensor &output_, int64_t
 }
 }
 
-Tensor log_softmax_cuda(const Tensor &input, const int64_t dim, const bool half_to_float){
-  return host_softmax<LogSoftMaxForwardEpilogue,true>(input, dim, half_to_float);
+TORCH_IMPL_FUNC(log_softmax_cuda_out) (
+  const Tensor &input,
+  const int64_t dim,
+  const bool half_to_float
+  const Tensor &output) {
+  auto res = host_softmax<LogSoftMaxForwardEpilogue,true>(input, dim, half_to_float);
+  output.copy_(res);
```

Comment:
So ideally I would refactor `host_softmax` so that it takes an `out` parameter but that function is also used by `softmax_cuda`, which is not structured. It can be, and the structure (no pun intended) of that kernel is very similar to that of `log_softmax`, so I think the port would be similar. However, it is not in any of the lists mentioned in #55070.

There are two options:
1. Port `_softmax` to structured so that this `host_softmax` function can be refactored to require an `out` parameter.
2. Do not port `_softmax` to structured. Add an optional `out` parameter to `host_softmax` and use that as the output `Tensor` if one is provided.

I prefer number 1. Let me know your thoughts.

Reply:
If not in a rush better to do the cluster of ports necessarily to put us in the better end state.

Reply:
I agree, but why is `_softmax` not on any of the lists in the first place?

Reply:
It's possible the end users are only reporting the top level functions, and not how they are implemented internally